### PR TITLE
Clean up multiprocess code

### PR DESCRIPTION
**Pull request by Xenorith**
- refactor commands launching the same process on multiple hosts into a single struct
- simplify the logic to ssh into different hosts with more straightforward handling of error channel
- move names.go under its own package to avoid import cycles

### CODE REVIEW

File: cli/src/alluxio.org/cli/processes/common.go [hunk 1]
```diff
@@ -184,35 +173,37 @@ func addStartFlags(argument string, cmd *env.StartProcessCommand) string {
...
-		hosts = append(hosts, masterList...)
-	}
-	if mode == "worker" || mode == "all" {
-		workerList, err := NewHostnamesFile("worker").getHostnames()
-		if err != nil {
-			return nil, stacktrace.Propagate(err, "cannot get workers")
+	for _, hostGroup := range hostGroups {
+		switch hostGroup {
+		case HostGroupMasters:
+		case HostGroupWorkers:
```
Comment (twalluxio) [inline cli/src/alluxio.org/cli/processes/common.go:197]: If the host group is 'masters' this part will do nothing. As a consequence, no masters can be found.

File: cli/src/alluxio.org/cli/processes/multi.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	return runSshCommand(
+		addStartFlags(cmd,
+			process.Service.Name,
+			cmd.Name,
+			p.SingleProcessName,
+		), p.HostnameFiles...)
+}
+
+func (p *MultiProcess) Stop(cmd *env.StopProcessCommand) error {
+	return runSshCommand(
```
Comment (twalluxio) [inline cli/src/alluxio.org/cli/processes/multi.go:94]: Here, the specified stop process doesn't have a name, so the stop command cannot be executed.

File: cli/src/alluxio.org/cli/processes/common.go [hunk 3]
```diff
@@ -184,35 +173,38 @@ func addStartFlags(argument string, cmd *env.StartProcessCommand) string {
...
-	}
-	if mode == "worker" || mode == "all" {
-		workerList, err := NewHostnamesFile("worker").getHostnames()
-		if err != nil {
-			return nil, stacktrace.Propagate(err, "cannot get workers")
+	for _, hostGroup := range hostGroups {
+		switch hostGroup {
+		case HostGroupMasters:
+			fallthrough
+		case HostGroupWorkers:
```
Comment (Kai-Zhang) [inline cli/src/alluxio.org/cli/processes/common.go:198]: maybe use this way:
```golang
switch hostGroup {
case HostGroupMasters, HostGroupWorkers:
  // hostname-related operations
}
```

**Review by twalluxio — APPROVED** [review]

**Review by twalluxio — APPROVED** [review]

**Comment by Xenorith** [comment]
alluxio-bot, merge this please

**Pull request merged**